packages/waku/package.json

```diff
@@ -67,11 +67,12 @@
   ],
   "scripts": {
     "dev": "swc src -d dist -w --strip-leading-paths",
-    "test": "vitest run",
+    "test": "vitest run && pnpm run test:types",
```

Comment:
oh on, this is weird to me, I thought type check should only in monorepo level

Reply:
do you mean the `tsc -b` on this line: https://github.com/dai-shi/waku/blob/main/package.json#L11 ?

does this actually type check anything? When I run it, theres no output even when I introduce ts errors in the project for checking its behavior.

Reply:
THis is what i'm expecting

```diff
Subject: [PATCH] fix: monorepo setup
---
Index: tsconfig.e2e.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tsconfig.e2e.json b/tsconfig.e2e.json
--- a/tsconfig.e2e.json	(revision 1469524a4ec329ce1c87db2a37088d2dbd4cce1d)
+++ b/tsconfig.e2e.json	(date 1728199013977)
@@ -4,7 +4,7 @@
     "outDir": "./dist/e2e",
     "types": ["node"]
   },
-  "include": ["playwright.config.base.ts", "playwright.config.ts", "./e2e"],
+  "include": ["playwright.config.base.ts", "playwright.config.ts"],
   "references": [
     {
       "path": "./e2e/fixtures/rsc-basic/tsconfig.json"
Index: tsconfig.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tsconfig.json b/tsconfig.json
--- a/tsconfig.json	(revision 1469524a4ec329ce1c87db2a37088d2dbd4cce1d)
+++ b/tsconfig.json	(date 1728199013983)
@@ -39,6 +39,10 @@
     // E2E
     {
       "path": "./tsconfig.e2e.json"
+    },
+    // Test
+    {
+      "path": "./packages/waku/tests"
     }
   ]
 }
Index: packages/waku/tests/tsconfig.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/packages/waku/tests/tsconfig.json b/packages/waku/tests/tsconfig.json
--- a/packages/waku/tests/tsconfig.json	(revision 1469524a4ec329ce1c87db2a37088d2dbd4cce1d)
+++ b/packages/waku/tests/tsconfig.json	(date 1728199060430)
@@ -1,8 +1,10 @@
 {
-  "extends": "../tsconfig.json",
+  "extends": "../../../tsconfig.json",
   "compilerOptions": {
-    "rootDir": "../",
-    "noEmit": true
+    "rootDir": "..",
+    "outDir": "./dist",
+    "tsBuildInfoFile": "./dist/.tsbuildinfo",
+    "lib": ["esnext"]
   },
   "include": ["."],
   "references": [

```

Reply:
1. monorepo `tsc -b` should handle cover the whole type check
2. test package.json should extends the global one, and do not have DOM because it's in vitest(node)

Reply:
got it, thanks! Updated to match this.